web/src/settings_components.ts

```diff
@@ -1800,12 +1800,10 @@ export function get_group_assigned_realm_permissions(group: UserGroup): {
                 assigned_permission_objects.push(assigned_permission_object);
             }
         }
-        if (assigned_permission_objects.length > 0) {
-            group_assigned_realm_permissions.push({
-                subsection_heading,
-                assigned_permissions: assigned_permission_objects,
-            });
-        }
+        group_assigned_realm_permissions.push({
+            subsection_heading,
+            assigned_permissions: assigned_permission_objects,
+        });
```

Comment:
Subsection headers are present in DOM even when there are no settings for that subsection so that order can be maintained during live-update when group gains a permission.

Reply:
For groups and streams, the new subsection is added at last. Maybe sorting by stream names would be better but not sure how to handle this without re-rendering the whole "Stream permissions" or "User group permissions" section.